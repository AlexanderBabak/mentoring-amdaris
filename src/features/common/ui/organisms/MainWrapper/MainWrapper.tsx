import { FC, PropsWithChildren } from "react";
import { Stack, StackProps } from "@mui/material";

const MainWrapper: FC<PropsWithChildren<{ bgcolor?: StackProps["bgcolor"] }>> = ({ bgcolor, children }) => (
  <Stack
    component="main"
    flex={1}
    overflow="auto"
    bgcolor={bgcolor ? bgcolor : (theme) => theme.palette.background.blue}
  >
    {children}
  </Stack>
);

export default MainWrapper;
