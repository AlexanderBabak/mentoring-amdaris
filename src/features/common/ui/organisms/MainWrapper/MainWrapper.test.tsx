import { render } from "@testing-library/react";
import MainWrapper from "./MainWrapper";

describe("MainWrapper", () => {
  it("should work as expected", async () => {
    const { baseElement } = render(
      <MainWrapper>
        <div data-testid="mock-Component" />
      </MainWrapper>,
    );

    expect(baseElement).toMatchSnapshot();
  });
  it("reflect background color by bgcolor props", async () => {
    const { baseElement } = render(
      <MainWrapper bgcolor={(theme) => theme.palette.primary.main}>
        <div data-testid="mock-Component" />
      </MainWrapper>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
