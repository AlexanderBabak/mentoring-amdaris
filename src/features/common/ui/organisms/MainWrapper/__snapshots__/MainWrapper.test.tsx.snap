// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainWrapper reflect background color by bgcolor props 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
  background-color: #1976d2;
}

<body>
  <div>
    <main
      class="MuiStack-root emotion-0"
    >
      <div
        data-testid="mock-Component"
      />
    </main>
  </div>
</body>
`;

exports[`MainWrapper should work as expected 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
}

<body>
  <div>
    <main
      class="MuiStack-root emotion-0"
    >
      <div
        data-testid="mock-Component"
      />
    </main>
  </div>
</body>
`;
